DiagnosticsExtension/Models/ConnectionStringValidator/EventHubsValidator.cs

```diff
@@ -101,5 +103,151 @@ protected async Task<TestConnectionData> TestConnectionStringAsync(string connec
 
             return data;
         }
+
+        async public Task<ConnectionStringValidationResult> ValidateViaAppsettingAsync(string appsettingName, string entityName)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                var result = await TestConnectionStringViaAppSettingAsync(appsettingName, entityName);
+                if (result.Succeeded)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Success;
+                }
+                else
+                {
+                    throw new Exception("Unexpected state reached: result.Succeeded == false is unexpected!");
+                }
+            }
+            catch (Exception e)
+            {
+                if (e is MalformedConnectionStringException)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.MalformedConnectionString;
+                }
+                else if (e.Message.Contains("managedidentitymissed"))
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Managedidentitymissed;
+                }
+                else if (e.Message.Contains("Unauthorized") || e.Message.Contains("unauthorized"))
+                {
+                    if (identityType == ConnectionStringValidationResult.ManagedIdentityType.User)
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.UserAssignedmanagedidentity;
+                    }
+                    else
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.SystemAssignedmanagedidentity;
+                    }
+                }
+                else if (e.Message.Contains("ManagedIdentityCredential"))
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredential;
```

Comment:
What issue is that status code convenying?

Reply:
fixed